spccore/exceptions.py

```diff
@@ -0,0 +1,77 @@
+import requests
+
+
+class SynapseError(Exception):
```

Comment:
extend cosntructor to hold `errorCode` https://docs.synapse.org/rest/org/sagebionetworks/repo/model/ErrorResponse.html

Reply:
For errors:
if response is only text, keep message as the text
if it is a JSON dict parse and check for "reason" key:
1. If key not exists, message is none
2. else message is the reason key